src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/populations/mutable/AggregateUniqueCounter.java

```diff
@@ -0,0 +1,44 @@
+package de.medizininformatikinitiative.fhir_data_evaluator.populations.mutable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Holds a set of unique {@link String}s.
+ * <p>
+ * This record is mutable because it holds a mutable {@link HashSet}, which is necessary to efficiently store and add
+ * many values without copying.
+ *
+ * @param aggregatedValues the set of unique aggregated values
+ */
+public record AggregateUniqueCounter(HashSet<String> aggregatedValues) {
+    public static final String EXTENSION_URL = "http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-aggregateMethod";
+    public static final String EXTENSION_VALUE = "unique-count";
+
+    public AggregateUniqueCounter {
+        aggregatedValues = new HashSet<>(aggregatedValues);
+    }
+
+    public static AggregateUniqueCounter empty() {
+        return new AggregateUniqueCounter(new HashSet<>());
+    }
+
+    public static AggregateUniqueCounter withValue(String value) {
+        return new AggregateUniqueCounter(new HashSet<>(Set.of(value)));
+    }
+
+    /**
+     * Mutates {@code aggregatedValues} to add a new value.
+     *
+     * @param val the value to add to the aggregated values
+     * @return itself with the mutated {@code aggregatedValues}
+     */
+    public AggregateUniqueCounter addValue(String val) {
+        aggregatedValues.add(val);
+        return this;
+    }
+
+    public int getScore() {
```

Comment:
please rename to score and return a Quantity here, because it's wrapped always into a Quantity